addons/weaponselect/functions/fnc_selectNextGrenade.sqf

```diff
@@ -62,14 +55,12 @@ private _vestGrenades =     vestItems     _unit select {_x in GVAR(GrenadesAll)
 private _backpackGrenades = backpackItems _unit select {_x in GVAR(GrenadesAll) && {_x != _nextGrenade}};
 
 // remove all grenades except those we are switching to --> this breaks the selector
-{_unit removeItemFromUniform  _x; false} count _uniformGrenades;
-{_unit removeItemFromVest     _x; false} count _vestGrenades;
-{_unit removeItemFromBackpack _x; false} count _backpackGrenades;
+{ _unit removeItem _x } forEach (magazines _unit) select {_x in GVAR(GrenadesAll) && {_x != _nextGrenade}};
```

Comment:
```suggestion
{ _unit removeItem _x } forEach ((magazines _unit) select {_x in GVAR(GrenadesAll) && {_x != _nextGrenade}});
```

`forEach` always need to have brakets around whole expression. Otherwise it will take only first part.